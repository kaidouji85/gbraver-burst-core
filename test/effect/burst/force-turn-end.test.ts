import {
  burst,
  EMPTY_ARMDOZER_EFFECT,
  EMPTY_ARMDOZER_STATE,
  EMPTY_GAME_STATE,
  EMPTY_PLAYER_STATE,
  GameState,
  PlayerState,
} from "../../../src";

/** バースト発動者 */
const burstPlayer: PlayerState = {
  ...EMPTY_PLAYER_STATE,
  playerId: "burst",
  armdozer: {
    ...EMPTY_ARMDOZER_STATE,
    enableBurst: true,
    battery: 1,
    maxBattery: 5,
    effects: [EMPTY_ARMDOZER_EFFECT, EMPTY_ARMDOZER_EFFECT],
    burst: {
      type: "ForceTurnEnd",
      recoverBattery: 2,
    },
  },
};

/** それ以外のプレイヤー */
const otherPlayer: PlayerState = { ...EMPTY_PLAYER_STATE, playerId: "other" };

/** ゲームに参加しているプレイヤー */
const players = [otherPlayer, burstPlayer];

test("相手ターンに強制ターンエンドを発動した場合、自分が攻撃プレイヤーになる", () => {
  const lastState: GameState = {
    ...EMPTY_GAME_STATE,
    activePlayerId: otherPlayer.playerId,
    players,
  };
  const result = burst(lastState, burstPlayer.playerId);
  expect(result).toMatchSnapshot();
});

test("自分ターンに強制ターンエンドを発動した場合、引き続き自分が攻撃プレイヤーである", () => {
  const lastState: GameState = {
    ...EMPTY_GAME_STATE,
    activePlayerId: burstPlayer.playerId,
    players,
  };
  const result = burst(lastState, burstPlayer.playerId);
  expect(result).toMatchSnapshot();
});
