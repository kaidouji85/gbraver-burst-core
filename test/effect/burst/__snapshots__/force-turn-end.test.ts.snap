// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`相手ターンに強制ターンエンドを発動した場合、自分が攻撃プレイヤーになる 1`] = `
{
  "activePlayerId": "burst",
  "effect": {
    "burst": {
      "recoverBattery": 2,
      "type": "ForceTurnEnd",
    },
    "burstPlayer": "burst",
    "name": "BurstEffect",
  },
  "players": [
    {
      "armdozer": {
        "battery": 5,
        "burst": {
          "recoverBattery": 5,
          "turnStartBatteryCorrect": 1,
          "type": "RecoverBattery",
        },
        "effects": [],
        "enableBurst": true,
        "hp": 3000,
        "id": "id",
        "maxBattery": 5,
        "maxHp": 3000,
        "name": "name",
        "power": 2000,
        "speed": 2000,
      },
      "pilot": {
        "enableSkill": true,
        "id": "EMPTY_PLAYER",
        "name": "名無し",
        "skill": {
          "recoverBattery": 2,
          "type": "RecoverBatterySkill",
        },
      },
      "playerId": "other",
    },
    {
      "armdozer": {
        "battery": 3,
        "burst": {
          "recoverBattery": 2,
          "type": "ForceTurnEnd",
        },
        "effects": [
          {
            "period": {
              "remainingTurn": 1,
              "type": "TurnLimit",
            },
            "type": "Empty",
          },
          {
            "period": {
              "remainingTurn": 1,
              "type": "TurnLimit",
            },
            "type": "Empty",
          },
        ],
        "enableBurst": false,
        "hp": 3000,
        "id": "id",
        "maxBattery": 5,
        "maxHp": 3000,
        "name": "name",
        "power": 2000,
        "speed": 2000,
      },
      "pilot": {
        "enableSkill": true,
        "id": "EMPTY_PLAYER",
        "name": "名無し",
        "skill": {
          "recoverBattery": 2,
          "type": "RecoverBatterySkill",
        },
      },
      "playerId": "burst",
    },
  ],
}
`;

exports[`自分ターンに強制ターンエンドを発動した場合、引き続き自分が攻撃プレイヤーである 1`] = `
{
  "activePlayerId": "burst",
  "effect": {
    "burst": {
      "recoverBattery": 2,
      "type": "ForceTurnEnd",
    },
    "burstPlayer": "burst",
    "name": "BurstEffect",
  },
  "players": [
    {
      "armdozer": {
        "battery": 5,
        "burst": {
          "recoverBattery": 5,
          "turnStartBatteryCorrect": 1,
          "type": "RecoverBattery",
        },
        "effects": [],
        "enableBurst": true,
        "hp": 3000,
        "id": "id",
        "maxBattery": 5,
        "maxHp": 3000,
        "name": "name",
        "power": 2000,
        "speed": 2000,
      },
      "pilot": {
        "enableSkill": true,
        "id": "EMPTY_PLAYER",
        "name": "名無し",
        "skill": {
          "recoverBattery": 2,
          "type": "RecoverBatterySkill",
        },
      },
      "playerId": "other",
    },
    {
      "armdozer": {
        "battery": 3,
        "burst": {
          "recoverBattery": 2,
          "type": "ForceTurnEnd",
        },
        "effects": [
          {
            "period": {
              "remainingTurn": 1,
              "type": "TurnLimit",
            },
            "type": "Empty",
          },
          {
            "period": {
              "remainingTurn": 1,
              "type": "TurnLimit",
            },
            "type": "Empty",
          },
        ],
        "enableBurst": false,
        "hp": 3000,
        "id": "id",
        "maxBattery": 5,
        "maxHp": 3000,
        "name": "name",
        "power": 2000,
        "speed": 2000,
      },
      "pilot": {
        "enableSkill": true,
        "id": "EMPTY_PLAYER",
        "name": "名無し",
        "skill": {
          "recoverBattery": 2,
          "type": "RecoverBatterySkill",
        },
      },
      "playerId": "burst",
    },
  ],
}
`;
